packages/manifold/contracts/physicalclaim/PhysicalClaimCore.sol

```diff
@@ -0,0 +1,337 @@
+     }
+
+    /**
+     * See {IPhysicalClaimCore-getPhysicalClaim}.
+     */
+    function getPhysicalClaim(uint256 instanceId) external override view returns(PhysicalClaim memory) {
+        return _getPhysicalClaim(instanceId);
+    }
+
+    /**
+     * Helper to get physical claim instance
+     */
+    function _getPhysicalClaim(uint256 instanceId) internal view returns(PhysicalClaim storage physicalClaimInstance) {
+        physicalClaimInstance = _physicalClaims[instanceId];
+        if (physicalClaimInstance.paymentReceiver == address(0)) {
+            revert InvalidInstance();
+        }
+    }
+
+    /**
+     * (Batch overload) see {IPhysicalClaimCore-burnRedeem}.
+     */
+    function burnRedeem(PhysicalClaimSubmission[] calldata submissions) external payable override nonReentrant {
+        if (submissions.length == 0) {
+            revert InvalidInput();
+        }
+        uint256 msgValueRemaining = msg.value;
+        for (uint256 i; i < submissions.length;) {
+            msgValueRemaining -= _burnRedeem(msgValueRemaining, submissions[i]);
+            unchecked { ++i; }
+        }
+
+        if (msgValueRemaining != 0) {
+            _forwardValue(payable(msg.sender), msgValueRemaining);
+        }
+    }
+
+    function _burnRedeem(uint256 msgValue, PhysicalClaimSubmission calldata submission) private returns (uint256) {
+        PhysicalClaim storage physicalClaimInstance = _getPhysicalClaim(submission.instanceId);
+
+        // Get the amount that can be burned
+        uint32 physicalClaimCount = _getAvailablePhysicalClaimCount(physicalClaimInstance.totalSupply, physicalClaimInstance.redeemedCount, submission.physicalClaimCount);
+
+        uint cost;
+        if (physicalClaimInstance.signer == address(0)) {
+            cost = 0;
+        } else {
+            // Check that the message value is what was signed...
+            _checkPriceSignature(submission.instanceId, submission.signature, submission.message, submission.nonce, physicalClaimInstance.signer, msgValue);
+        }
+
+        if (physicalClaimCount > 1) {
+            cost *= physicalClaimCount;
+        }
+        if (cost > msgValue) {
+            revert InvalidPaymentAmount();
+        }
+        if (cost > 0) {
+            _forwardValue(physicalClaimInstance.paymentReceiver, cost);
+        }
+
+        // Do physical claim
+        _burnTokens(physicalClaimInstance, submission.burnTokens, physicalClaimCount, msg.sender, submission.data);
+        _redeem(submission.instanceId, physicalClaimInstance, msg.sender, submission.physicalClaimCount, submission.variation, submission.data);
+
+        return cost;
+    }
+
+    function _checkPriceSignature(uint256 instanceId, bytes calldata signature, bytes32 message, bytes32 nonce, address signingAddress, uint cost) internal {
+        // Verify valid message based on input variables
+        bytes32 expectedMessage = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", instanceId, cost));
```

Comment:
See, here, you are actually passing in the remainingValue, so this prob doesn't work...

Reply:
Updated